### PR TITLE
Homework 29-04-24

### PR DESCRIPTION
**Pull request by VahagnSyan**
There are Employee.sql file, where Hospital queries performed.

### CODE REVIEW

File: Vahag/databases/29-04-2024/Employee.sql [hunk 1]
```diff
@@ -0,0 +1,38 @@
+use Hospital;
+
+--1
+select h.address as H_address, h.name as H_name,
```
Comment (elya-im) [inline Vahag/databases/29-04-2024/Employee.sql]: You can use the aliases with spaces like this, this version is more readable
```
select h.address as "Hospital Address"
           ,h.name as "Hospital name"
           ,d.name as "Department name"
           ,dc.name as "Doctor name"
           , p.name as "Patient name" 
```

File: Vahag/databases/29-04-2024/Employee.sql [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+inner join Patient p on p.id = d.hospital_id
+inner join Invoice i on i.patient_id = p.id
+where i.description = 'Doctor Visit';
+
+--2
+select p.name, p.phone_number
+from Patient p
+inner join Prescription pr
+on pr.patient_id = p.id
+where pr.prescription_date like '%-04-%';
```
Comment (elya-im) [inline Vahag/databases/29-04-2024/Employee.sql]: use `MONTH()` function to get the month for the date

File: Vahag/databases/29-04-2024/Employee.sql [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+d.name as D_name, doc.name as Doc_name, p.name as P_name 
+from Hospital h
+inner join Department d on h.id = d.hospital_id
+inner join Doctor doc on doc.department_id = d.id
+inner join Patient p on p.id = d.hospital_id
+inner join Invoice i on i.patient_id = p.id
+where i.description = 'Doctor Visit';
+
+--2
+select p.name, p.phone_number
```
Comment (elya-im) [inline Vahag/databases/29-04-2024/Employee.sql]: use meaningful aliases for the selected columns to make the result readable

File: Vahag/databases/29-04-2024/Employee.sql [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+from Patient p
+inner join Prescription pr
+on pr.patient_id = p.id
+where pr.prescription_date like '%-04-%';
+
+--3
+select d.name, d.specialization
```
Comment (elya-im) [inline Vahag/databases/29-04-2024/Employee.sql]: use meaningful aliases for the selected columns to make the result readable

File: Vahag/databases/29-04-2024/Employee.sql [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+from Doctor d
+inner join Department dep
+on d.department_id = dep.id
+where dep.id = 1; -- can change department id there
+
+--4
+select d.name as D_name, doc.name as Doc_name, doc.specialization
```
Comment (elya-im) [inline Vahag/databases/29-04-2024/Employee.sql]: use meaningful aliases for the selected columns to make the result readable

File: Vahag/databases/29-04-2024/Employee.sql [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+inner join Prescription pr
+on pr.patient_id = p.id
+where pr.prescription_date like '%-04-%';
+
+--3
+select d.name, d.specialization
+from Doctor d
+inner join Department dep
+on d.department_id = dep.id
+where dep.id = 1; -- can change department id there
```
Comment (elya-im) [inline Vahag/databases/29-04-2024/Employee.sql]: there is no need to specify the department, all the doctors should selected with their department name

File: Vahag/databases/29-04-2024/Employee.sql [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+
+--4
+select d.name as D_name, doc.name as Doc_name, doc.specialization 
+from Department d
+left join Doctor doc
+on d.id = doc.department_id;
+
+--5
+select p.name, p.phone_number, p.address, p.date_of_birth, ph.name
```
Comment (elya-im) [inline Vahag/databases/29-04-2024/Employee.sql]: use meaningful aliases for the selected columns to make the result readable

File: Vahag/databases/29-04-2024/Employee.sql [hunk 8]
```diff
@@ -0,0 +1,38 @@
+use Hospital;
+
+--1
+select h.address as H_address, h.name as H_name, 
+d.name as D_name, doc.name as Doc_name, p.name as P_name 
+from Hospital h
+inner join Department d on h.id = d.hospital_id
+inner join Doctor doc on doc.department_id = d.id
+inner join Patient p on p.id = d.hospital_id
```
Comment (elya-im) [inline Vahag/databases/29-04-2024/Employee.sql]: `p.id = d.hospital_id` is an incorrect comparison, as they are not the same column

File: Vahag/databases/29-04-2024/Employee.sql (hunk 1 above)
Comment (VahagnSyan) [reply Vahag/databases/29-04-2024/Employee.sql]: Done:)

File: Vahag/databases/29-04-2024/Employee.sql (hunk 2 above)
Comment (VahagnSyan) [reply Vahag/databases/29-04-2024/Employee.sql]: Done:)

File: Vahag/databases/29-04-2024/Employee.sql (hunk 3 above)
Comment (VahagnSyan) [reply Vahag/databases/29-04-2024/Employee.sql]: Done:)

File: Vahag/databases/29-04-2024/Employee.sql (hunk 4 above)
Comment (VahagnSyan) [reply Vahag/databases/29-04-2024/Employee.sql]: Done:)

File: Vahag/databases/29-04-2024/Employee.sql (hunk 5 above)
Comment (VahagnSyan) [reply Vahag/databases/29-04-2024/Employee.sql]: Done:)

File: Vahag/databases/29-04-2024/Employee.sql (hunk 6 above)
Comment (VahagnSyan) [reply Vahag/databases/29-04-2024/Employee.sql]: Done:)

File: Vahag/databases/29-04-2024/Employee.sql (hunk 7 above)
Comment (VahagnSyan) [reply Vahag/databases/29-04-2024/Employee.sql]: Done:)

File: Vahag/databases/29-04-2024/Employee.sql (hunk 8 above)
Comment (VahagnSyan) [reply Vahag/databases/29-04-2024/Employee.sql]: Done:)

**Review by elya-im — APPROVED** [review]

**Pull request merged**